### DISCUSSION TITLE
Log Struct(s) Management - app, transfer, command, etc...

### DISCUSSION THREAD

**Discussion by fntyler** [drakkan/sftpgo#1202] [Q&A] [answered]
Is there a feature/configuration parameter to control or organize each Log struct stream? For example, if using Docker environment variables:
```
...
environment:
   SFTPGO_LOG_APP_FILE_PATH: /path/to/logs/logfilename-app.log
...
```
Referencing, [logs.md](https://github.com/drakkan/sftpgo/blob/main/docs/logs.md)

The goal being to have the following: `sftpgo-{app,transfer,command}.log`

Is there a feature for sending Log structs to a database, similarly to the way that Event Manager can be configured? I.e., Event Manager -> sftpgo-plugin-eventstore

* Obviously, plugins are leveraged here.

---
My Setup:
Docker Image: `drakkan/sftpgo:v2.4.4-plugins`
Log Management: Docker env. variables

**Comment by drakkan** [comment, marked as answer]
Hello, the logs are written to a single file (on Docker they are written to the stdout by default). The eventstore plugin is enough to persist transfer and provider logs to a supported database

**Reply by fntyler** [reply]
Thank you for the reply. I am aware that the Docker container logs are written to stdout by default. I was to inquiring about additional configuration options for managing the individual Log Structs' stdout streams. The goal being to allow for each Log Struct to be written to a dedicated log file rather than just one (sftpgo.log or stdout). Could this be a possible feature that is implemented in the future?

**Reply by drakkan** [reply]
> Thank you for the reply. I am aware that the Docker container logs are written to stdout by default. I was to inquiring about additional configuration options for managing the individual Log Structs' stdout streams. The goal being to allow for each Log Struct to be written to a dedicated log file rather than just one (sftpgo.log or stdout). Could this be a possible feature that is implemented in the future?

maybe, I'm not sure. Anyway you can easily parse the log file if you want to import the logs yourself in your system. I think it could make sense separe application logs, but not commands and transfer logs. Command and transfer logs can already be saved to a database using plugins

**Reply by fntyler** [reply]
> I think it could make sense separe application logs, but not commands and transfer logs

I concur.

If it comes to it, I'll attempt parsing them myself. Thank you for the feedback and quick replies. Marking as answered.
